src/clarity/datagrid/datagrid-row.ts

```diff
@@ -16,7 +16,8 @@ import {Selection} from "./providers/selection";
     `,
     host: {
         "[class.datagrid-row]": "true",
-        "[class.datagrid-selected]": "selected"
+        "[class.datagrid-selected]": "selected",
+        "(click)": "toggleSelection($event.target)"
```

Comment:
You are adding a click handler on an inert element, which screenreaders will ignore.
For this to be accessible, you will need to add all the aria classes needed to describe the behavior you are implementing.